### ISSUE TITLE
raster extension and trouble opening the test data

### ISSUE THREAD

**Issue by schwehr** [stac-utils/pystac#383]
This is in a custom bazel based build on a non-public linux distribution, so there could be compounding issues.  It's using this python:

```
sys.version_info(major=3, minor=6, micro=13, releaselevel='final', serial=0)
```

I'm sometimes seeing a failure from this string from https://github.com/stac-utils/pystac/blob/6f4ca38d28eb1971b94e4517049c6d5af52d0672/tests/data-files/raster/raster-planet-example.json#L49

```
          "unit": "W⋅sr−1⋅m−2⋅nm−1",
```

But running in two different environments I get different results.  locally it works.  But in an isolated test CI forge it fails.

This is the failure:

```
=================================== FAILURES ===================================
_______________________ RasterTest.test_validate_raster ________________________

self = <pystac.tests.extensions.test_raster.RasterTest testMethod=test_validate_raster>

    def test_validate_raster(self) -> None:
>       item = pystac.read_file(self.PLANET_EXAMPLE_URI)

tests/extensions/test_raster.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
__init__.py:89: in read_file
    return STACObject.from_file(href)
stac_object.py:477: in from_file
    o = stac_io.read_stac_object(href)
stac_io.py:137: in read_stac_object
    d = self.read_json(source)
stac_io.py:116: in read_json
    txt = self.read_text(source)
stac_io.py:180: in read_text
    return self.read_text_from_href(href, *args, **kwargs)
stac_io.py:192: in read_text_from_href
    return f.read()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <encodings.ascii.IncrementalDecoder object at 0x7f412b4d1eb8>
input = b'{\n  "stac_extensions": [\n    "https://stac-extensions.github.io/sat/v1.0.0/schema.json",\n    "https://stac-extens...id": 857\n  },\n  "bbox": [\n    119.749003733,\n    -0.902255498348,\n    119.990422256,\n    -0.782842043003\n  ]\n}'
final = True

    def decode(self, input, final=False):
>       return codecs.ascii_decode(input, self.errors)[0]
E       UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1417: ordinal not in range(128)

<embedded stdlib>/encodings/ascii.py:26: UnicodeDecodeError
```

**Comment by duckontheweb** [comment]
@schwehr I think this may be fixed by #361. I was seeing similar problems in the Windows CI builds that were related to the fact that Windows doesn't use UTF-8 as the default encoding. In that PR I added explicit `encoding="utf-8"` arguments to all of the `open` calls for reading and writing.

**Comment by duckontheweb** [comment]
@schwehr Have you been able to verify if #361 fixes this?

**Comment by schwehr** [comment]
I think so, but not 100% sure.  I'm part way through catching up to upstream head.  I should be able to check later today or tomorrow.

**Comment by duckontheweb** [comment]
Sorry to bug you again @schwehr, but I'm wondering if we can close this.

**Comment by schwehr** [comment]
Yes.  We can close it.  Sorry I didn't follow up sooner

**Comment by duckontheweb** [comment]
Closed via #361
